lib/mirage/client/runner.rb

```diff
@@ -42,12 +42,19 @@ def stop options={:port => []}
     #   Mirage.running? :port => port -> boolean indicating whether Mirage is running on *locally* on the given port
     #   Mirage.running? url -> boolean indicating whether Mirage is running on the given URL
     def running? options_or_url = {:port => 7001}
-      url = options_or_url.kind_of?(Hash) ? "http://localhost:#{options_or_url[:port]}" : options_or_url
+      if options_or_url.kind_of?(Hash)
+        if options_or_url[:url]
+          url = options_or_url[:url]
+        else
+          url = "http://localhost:#{options_or_url[:port]}"
+        end
+      else
+        url = options_or_url
+      end
```

Comment:
It's getting really complex and hard to read. Could we move it to separated private method to determine the `url` value and use guard clauses there?

```
...

def running?(options_or_url = {:port => 7001})
  HTTParty.get(mirage_url(options_or_url))
  true
rescue Errno::ECONNREFUSED
  false
end

private

def mirage_url(options_or_url)
  return options_or_url unless options_or_url.kind_of?(Hash)
  return options_or_url[:url] if options_or_url[:url]
  "http://localhost:#{options_or_url[:port]}"
end
```